client/context/context.go

```diff
@@ -93,174 +88,90 @@ func NewCLIContext() CLIContext {
-	}
-	if home == "" {
-		errMsg.WriteString("--home ")
-	}
-	if nodeURI == "" {
-		errMsg.WriteString("--node ")
-	}
-	if errMsg.Len() != 0 {
-		fmt.Printf("Must specify these options: %s when --trust-node is false\n", errMsg.String())
-		os.Exit(1)
-	}
-
-	node := rpcclient.NewHTTP(nodeURI, "/websocket")
-	cacheSize := 10 // TODO: determine appropriate cache size
-	verifier, err := tmliteProxy.NewVerifier(
-		chainID, filepath.Join(home, ".gaialite"),
-		node, log.NewNopLogger(), cacheSize,
-	)
-
-	if err != nil {
-		fmt.Printf("Create verifier failed: %s\n", err.Error())
-		fmt.Printf("Please check network connection and verify the address of the node to connect to\n")
-		os.Exit(1)
-	}
-
-	return verifier
+func NewCLIContextTx(cdc *codec.Codec) *CLIContext {
+	ctx := NewCLIContext(cdc).SetAccountDecoder()
+	ctx.PrepareTxBldrOffline()
+	return ctx
 }
 
-func fromFields(from string) (fromAddr types.AccAddress, fromName string) {
-	if from == "" {
-		return nil, ""
-	}
+// PrepareTxBldrOffline sets the transaction builder for the context w/o
+// setting the sequence or account numbers
+func (ctx *CLIContext) PrepareTxBldrOffline() {
+	ctx.TxBldr = authtxb.NewTxBuilderFromCLI().WithTxEncoder(GetTxEncoder(ctx.Codec))
+}
 
-	keybase, err := keys.GetKeyBase()
-	if err != nil {
-		fmt.Println("no keybase found")
-		os.Exit(1)
+// PrepareTxBldrWithAddress looks up the acc and seq numbs for an addr, also
+// ensuring it exists
+func (ctx *CLIContext) PrepareTxBldrWithAddress(addr sdk.AccAddress) error {
+	if err := ctx.EnsureAccountExists(addr); err != nil {
+		return err
 	}
 
-	var info cskeys.Info
-	if addr, err := types.AccAddressFromBech32(from); err == nil {
-		info, err = keybase.GetByAddress(addr)
+	if ctx.TxBldr.GetAccountNumber() == 0 || ctx.TxBldr.GetSequence() == 0 {
+		accNum, seq, err := ctx.FetchAccAndSeqNums(addr)
 		if err != nil {
-			fmt.Printf("could not find key %s\n", from)
-			os.Exit(1)
-		}
-	} else {
-		info, err = keybase.Get(from)
-		if err != nil {
-			fmt.Printf("could not find key %s\n", from)
-			os.Exit(1)
+			return err
 		}
+		ctx.TxBldr = ctx.TxBldr.WithAccountNumber(accNum).WithSequence(seq)
 	}
-
-	fromAddr = info.GetAddress()
-	fromName = info.GetName()
-	return
+	return nil
 }
 
-// WithCodec returns a copy of the context with an updated codec.
-func (ctx CLIContext) WithCodec(cdc *codec.Codec) CLIContext {
-	ctx.Codec = cdc
+// SetAccountDecoder returns a copy of the context with an updated account
+// decoder.
+func (ctx *CLIContext) SetAccountDecoder() *CLIContext {
+	ctx.AccDecoder = ctx.GetAccountDecoder()
 	return ctx
 }
 
 // GetAccountDecoder gets the account decoder for auth.DefaultAccount.
-func GetAccountDecoder(cdc *codec.Codec) auth.AccountDecoder {
+func (ctx *CLIContext) GetAccountDecoder() auth.AccountDecoder {
 	return func(accBytes []byte) (acct auth.Account, err error) {
-		err = cdc.UnmarshalBinaryBare(accBytes, &acct)
+		err = ctx.Codec.UnmarshalBinaryBare(accBytes, &acct)
```

Comment:
```golang
if err := ctx.Codec.UnmarshalBinaryBare(accBytes, &acct); err != nil {
    // TODO: remove this, and return the error,
    // but first figure out where this is used
    panic(err)
}
return acct, nil
```

Reply:
I'm also going to modify this to return an error instead of panic. Good call on the better syntax here @alessio